### DISCUSSION TITLE
Should the email be used to identify the user when connecting with Dex ?

### DISCUSSION THREAD

**Discussion by Calvinaud** [litmuschaos/litmus#4519] [General]
## Question

Hello,

My question is around the way the user record is retrieve when connection through Dex.
At the moment, the `email` retrieve from the token is used to retrieve the record corresponding to the user in the DB:

In https://github.com/litmuschaos/litmus/blob/master/litmus-portal/authentication/pkg/user/repository.go#L48 it use the `UserName` which take the value of the email here: https://github.com/litmuschaos/litmus/blob/master/chaoscenter/authentication/api/handlers/rest/dex_auth_handler.go#L127

When looking at some documentation it's often not recommended to use the email for this (as I understand) since it's subject to change and not sure to be unique:
- In google [documentation](https://developers.google.com/identity/openid-connect/openid-connect): 
![image](https://github.com/litmuschaos/litmus/assets/26198890/16b9591b-91c3-4d25-a8f4-f6fba5cb3aef)
- In [Azure](https://learn.microsoft.com/en-us/entra/identity-platform/claims-validation#validate-the-subject):
![image](https://github.com/litmuschaos/litmus/assets/26198890/6ed87e4d-626b-40e5-befe-84a5591d065a)

Should the way the record is retrieved be changed so it using the `sub` claim instead of the `email` ?

**Comment by SarthakJain26** [comment]
Looks to be a good suggestion, we can take this. @Calvinaud would you like to contribute to this?

**Comment by SarthakJain26** [comment]
@DarthBenro008 would you like to share your views on this

**Comment by DarthBenro008** [comment]
Thanks for pointing this out @Calvinaud !

It is indeed true that its not the best practice and we need to use `sub` instead of `email`. But we have implemented email because, folks who logged in via dex can also login via normal auth. If we use `sub`, we loose this data correlation. 



If the following cases are true, we can switch to `sub` from `email` for username.
- users signed up via Dex are not allowed to login via normal auth
- Dex users cannot access LitmusUI if admin opts to switch off Dex integration.

@SarthakJain26 can you please confirm if that's the case?

**Comment by SarthakJain26** [comment]
@DarthBenro008 I think this should be fine. We can separate out accounts logged in via dex and normal logged in accounts. And based on this Dex users will not be able to access LitmusUI if admin opts to switch off Dex integration, which is an expected behaviour. 
@Calvinaud any further thoughts on this?

**Reply by Calvinaud** [reply]
Hello,
Not really. It seem fine for me that the authentication methods are not linked.

**Reply by SarthakJain26** [reply]
@Calvinaud / @DarthBenro008 any interests on picking this up?
